docs/guides/runtime_tracing.md

```diff
@@ -0,0 +1,27 @@
+# Runtime Tracing
+
+The Crystal runtime has a tracing feature for low level functionality. It prints diagnostic info about runtime internals.
+
+A program must be built with the flag `-Dtracing` to support tracing.
+At runtime, the individual tracing components can be enabled via the environment variable `CRYSTAL_TRACE`. It receives a comma separated list of sections to enable.
+
+- `CRYSTAL_TRACE=none` Disable tracing (default)
+- `CRYSTAL_TRACE=gc`: Enable tracing for the garbage collector
+- `CRYSTAL_TRACE=sched`: Enalbe tracing for the scheduler
+- `CRYSTAL_TRACE=gc,sched`: Enable tracing for the garbage collector and scheduler
+- `CRYSTAL_TRACE=all` Enable all tracing
+
+Example:
+
+```console
+$ crystal build -Dtracing hello-world.cr
+$ CRYSTAL_TRACE=sched ./hello-world
+sched.spawn 70569399740240 thread=0x7f48d7dc9740:? fiber=0x7f48d7cd0f00:main fiber=0x7f48d7cd0dc0:Signal Loop
+sched.enqueue 70569399831716 thread=0x7f48d7dc9740:? fiber=0x7f48d7cd0f00:main fiber=0x7f48d7cd0dc0:Signal Loop duration=163
```

Comment:
I think it would be nice to expand upon the structure of the output. Like `duration=163`; what unit is that? Probably could do that as a follow up tho.

Reply:
Yes, we should, all times and durations are in nanoseconds!

Reply:
I second that it's missing what output to expect

Reply:
Especially since a 5 seconds sleep will print 5000000000 :scream: